tools/buildtools/topologyeditors/add_top.xml

```diff
@@ -0,0 +1,129 @@
+            &&
+            python3 $__tool_directory__/newmolparameters.py --top_file 'odoylerules' --atom_file '$functionality.nonbondparam' --mol_file '$functionality.bondparam' --out '$output1'
+        #end if
+
+    ]]></command>
+    <inputs>
+        
+        <conditional name="functionality">
+             <param name="what2add" type="select" label="What new information are you adding to your topology file?">
+                 <option value="mol">A molecule's parameters</option>
+                 <option value="restraints">Position restraint information</option>
+                 <option value="both">Both a molecule's topology information and position restrain information</option>
+             </param>
+             <when value="mol">
+                 <param name="inputtopology" type="data" format="top" label="GROMACS Topology file to be modified." help="Topology file with missing information"/>
+                 <param name="nonbondparam" type="data" format="txt" label="Atom Types with Nonbonded Parameters" help="Atom types information from grep"/>
+                 <param name="bondparam" type="data" format="txt,itp" label="Molecule type information with bonded parameters" help="molecule type information"/>
+             </when>
+             <when value="restraints">
+                 <param name="inputtopology" type="data" format="top" label="GROMACS Topology file to be modified" help="Topology file with missing information"/>
+                 <param name="posres" type="data" format="itp" label="position restraint file" help="Position restraint file created previously"/>
+		 <param name="targetmolecule" type="text" label="Target molecule type" help="The molecule type name to which the restraints were applied"/>
+             </when>
+             <when value="both">
+                 <param name="inputtopology" type="data" format="top" label="GROMACS Topology file to be modified" help="Topology file with missing information"/>
+                 <param name="nonbondparam" type="data" format="txt" label="Atom Types with Nonbonded Parameters" help="Atom types information from grep"/>
+                 <param name="bondparam" type="data" format="txt,itp" label="Molecule type information with bonded parameters" help="molecule type information"/>
+                 <param name="posres" type="data" format="itp" label="position restraint file" help="Position restraint file created previously"/>
+                 <param name="targetmolecule" type="text" label="Target molecule type" help="The molecule type name to which the restraints were applied"/>
+             </when>
+
+        </conditional>  
+    </inputs>
+    <outputs>
+        <data name="output1" format="top" label="Modified file with new topology information ${on_string}">
+        </data>
+    </outputs>
+    <tests>
+        <test>
+            <param name="functionality|what2add" value="mol" />
```

Comment:
can you use the more explicit synatx here:

<conditional name="">
    <param name="what2add" value="mol" />
    <param name="inputtopology" value="cid1_GMX.top" />
    ....
</conditional>

Reply:
not sure I understand what you are referring to

Reply:
?

Reply:
fixed